File: app/src/main/java/com/lumiyaviewer/lumiya/render/glres/GLResource.java
package com.lumiyaviewer.lumiya.render.glres;

public abstract class GLResource implements GLGenericResource {
    public final int handle;

    public GLResource(GLResourceManager glResourceManager) {
        this.handle = Allocate(glResourceManager);
    }

    protected abstract int Allocate(GLResourceManager glResourceManager);
}

File: app/src/main/java/com/lumiyaviewer/lumiya/render/glres/GLResource.kt
package com.lumiyaviewer.lumiya.render.glres

abstract class GLResource(manager: GLResourceManager) : GLGenericResource {
    @JvmField val handle: Int = Allocate(manager)

    @Suppress("FunctionName")
    protected abstract fun Allocate(manager: GLResourceManager): Int
}
